JustSaying/AwsTools/MessageHandling/SnsTopicByName.cs

```diff
@@ -96,5 +97,68 @@ public async Task<bool> CreateAsync()
 
             return false;
         }
+
+        public async Task<bool> CreateWithEncryptionAsync(ServerSideEncryption config)
+        {
+            var created = await CreateAsync().ConfigureAwait(false);
+
+            ServerSideEncryption = await ExtractServerSideEncryptionFromTopicAttributes().ConfigureAwait(false);
+
+            await EnsureServerSideEncryptionIsUpdatedAsync(config).ConfigureAwait(false);
+
+            return created;
+        }
+
+        private async Task<ServerSideEncryption> ExtractServerSideEncryptionFromTopicAttributes()
+        {
+            var attributesResponse = await Client.GetTopicAttributesAsync(Arn).ConfigureAwait(false);
+            
+            if (!attributesResponse.Attributes.TryGetValue(JustSayingConstants.AttributeEncryptionKeyId, out var encryptionKeyId))
+            {
+                return null;
+            }
+
+            return new ServerSideEncryption
+            {
+                KmsMasterKeyId = encryptionKeyId
+            };
+        }
+
+        private async Task EnsureServerSideEncryptionIsUpdatedAsync(ServerSideEncryption config)
+        {
+            if (ServerSideEncryptionNeedsUpdating(config))
+            {
+                var request = new SetTopicAttributesRequest
+                {
+                    TopicArn = Arn,
+                    AttributeName = JustSayingConstants.AttributeEncryptionKeyId,
+                    AttributeValue = config?.KmsMasterKeyId ?? string.Empty
+                };
+
+                var response = await Client.SetTopicAttributesAsync(request).ConfigureAwait(false);
+
+                if (response.HttpStatusCode == HttpStatusCode.OK)
```

Comment:
We seem to be ignoring any HTTP errors here, they're not even being logged, which seems incorrect.

Reply:
I used the same pattern that is there for the sqs queue, which also doesn't fail if the request to set the attributes fails, but agree it shouldn't simply ignore it, so added the logging